src/mdframe/analysis.py

```diff
@@ -0,0 +1,140 @@
+from pathlib import Path
+from typing import Dict, Any, Tuple, Optional, Literal, get_args, List
+from reader import Config, run
+import pandas as pd
+import matplotlib.pyplot as plt
+
+config = Config(
+    data_dir_path=Path(__file__).parent / "data_dir", 
+    metadata_file_extension="toml",
+    schema_loc=Path('schema.json')
+)
+
+def load_flattened_data(config):
```

Comment:
Could you also write unit tests for each of the functions in this file? We would like to test everything atomically and we're preparing for some continuous integration in the near future.

Reply:
I've written unit tests for all functions except `generate_histogram`...still thinking about how I could potentially write tests for `generate_histogram` since it renders a pyplot and doesn't have any explicit output.